colcon_cargo/package_identification/cargo.py

```diff
@@ -33,8 +34,11 @@ def identify(self, metadata):  # noqa: D102
         metadata.type = 'cargo'
         if metadata.name is None:
             metadata.name = data['name']
-        metadata.dependencies['build'] |= data['depends']
-        metadata.dependencies['run'] |= data['depends']
+
+        dependencies = set(DependencyDescriptor(name)
+                           for name in data['depends'])
+        metadata.dependencies['build'] |= dependencies
+        metadata.dependencies['run'] |= dependencies
```

Comment:
With this patch both dependencies are sharing the same `DependencyDescriptor` instance. I would suggest to use separate ones instead:

```
metadata.dependencies['build'] |= {
    DependencyDescriptor(name) for name in data['depends']}
metadata.dependencies['run'] |= {
    DependencyDescriptor(name) for name in data['depends']}
```